tests/Properties.hs

```diff
@@ -82,36 +91,54 @@ prop_lines_lazy2 =
+
+-- Ensure that readWord64 and readInteger over lazy ByteStrings are not
 -- excessively strict.
-prop_readIntSafe         = (fst . fromJust . D.readInt) (Chunk (C.pack "1z") Empty)         == 1
-prop_readIntUnsafe       = (fst . fromJust . D.readInt) (Chunk (C.pack "2z") undefined)     == 2
+prop_readWordSafe        = (fst . fromJust . D.readWord64) (Chunk (C.pack "1z") Empty)      == 1
+prop_readWordUnsafe      = (fst . fromJust . D.readWord64) (Chunk (C.pack "2z") undefined)  == 2
 prop_readIntegerSafe     = (fst . fromJust . D.readInteger) (Chunk (C.pack "1z") Empty)     == 1
 prop_readIntegerUnsafe   = (fst . fromJust . D.readInteger) (Chunk (C.pack "2z") undefined) == 2
-prop_readIntBoundsCC     = let !smax   = show (maxBound :: Int)
-                               !smin   = show (minBound :: Int)
-                               !smax1  = show (fromIntegral (maxBound :: Int) + 1 :: Integer)
-                               !smin1  = show (fromIntegral (minBound :: Int) - 1 :: Integer)
-                               !smax10 = show (fromIntegral (maxBound :: Int) + 10 :: Integer)
-                               !smin10 = show (fromIntegral (minBound :: Int) - 10 :: Integer)
-                            --
-                            in C.readInt (spack smax) == good maxBound
-                            && C.readInt (spack smin) == good minBound
-                            --
-                            && C.readInt (spackPlus smax) == good maxBound
-                            && C.readInt (spackMinus smax) == good (negate maxBound)
-                            --
-                            && C.readInt (spackZeros smax) == good maxBound
-                            && C.readInt (spackZeros smin) == good minBound
-                            --
-                            && C.readInt (spack smax1 ) == Nothing
-                            && C.readInt (spack smin1 ) == Nothing
-                            --
-                            && C.readInt (spack smax10) == Nothing
-                            && C.readInt (spack smin10) == Nothing
-                            --
-                            && C.readInt (spackLong smax) == Nothing
-                            && C.readInt (spackLong smin) == Nothing
+prop_readNaturalSafe     = (fst . fromJust . D.readNatural) (Chunk (C.pack "1z") Empty)     == 1
+prop_readNaturalUnsafe   = (fst . fromJust . D.readNatural) (Chunk (C.pack "2z") undefined) == 2
+prop_readIntBoundsCC     =     rdWordBounds @Word
+                            && rdWordBounds @Word8
+                            && rdWordBounds @Word16
+                            && rdWordBounds @Word32
+                            && rdWordBounds @Word64
+                            && rdIntBounds  @Int
+                            && rdIntBounds  @Int8
+                            && rdIntBounds  @Int16
+                            && rdIntBounds  @Int32
+                            && rdIntBounds  @Int64
```

Comment:
It seems that combining properties with `&&` will make them a bit hard to debug if they do end up failing. Combining them with [`(.&&.)`](https://hackage.haskell.org/package/QuickCheck-2.14.2/docs/Test-QuickCheck.html#v:.-38--38-.) should make that easier, I think.

At this stage, it's probably not worth changing though.

Reply:
I did not know about ".&&.", improving the tests without touching the main code may be reasonable?  Your call...

Reply:
I'd prefer to get this PR finished as soon as possible. Feel free to send more improvements in follow-up PRs.

Reply:
> I'd prefer to get this PR finished as soon as possible. Feel free to send more improvements in follow-up PRs.

Makes sense.  Thanks!

Reply:
I think it is a good use case for [`conjoin`](https://hackage.haskell.org/package/QuickCheck-2.14.2/docs/Test-QuickCheck.html#v:conjoin).